lib/plausible/stats/dashboard_query_serializer.ex

```diff
@@ -0,0 +1,90 @@
+defmodule Plausible.Stats.DashboardQuerySerializer do
+  @moduledoc """
+  Takes a `%ParsedQueryParams{}` struct and turns it into a query
+  string.
+  """
+
+  alias Plausible.Stats.{ParsedQueryParams, DashboardQueryParser}
+
+  def serialize(%ParsedQueryParams{} = params) do
+    encoded_query =
+      params
+      |> Map.to_list()
+      |> Enum.flat_map(&get_serialized_fields/1)
+      |> Enum.sort_by(&elem(&1, 0))
+      |> Enum.map_join("&", fn {key, value} -> "#{key}=#{value}" end)
```

Comment:
https://hexdocs.pm/elixir/1.19.4/URI.html#encode_query/2 accepts enumerables too, not only maps, so I think:

```suggestion
      |> URI.encode_query()
```

should be equivalent?